File: Quiz.Entity/Models/Exam.cs
using System.ComponentModel.DataAnnotations;

namespace Quiz.Entity.Models
{
    public class Exam : BaseModel
    {
        [Key]
        public int ExamId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsRandom { get; set; }
        public decimal PassScore { get; set; }
        public ExamType ExamType { get; set; }
        public DateTime? StartTime { get; set; }
        public DateTime? EndTime { get; set; }

        public virtual ICollection<Question> Questions { get; set; }
        public virtual ICollection<UserExam> UserExams { get; set; }
    }

    public enum ExamType
    {
        Timed,
        Untimed,
        Practice
    }
}

File: Quiz.Android/lib/models/exam.dart
class Exam {
  final int examId;
  final String name;
  final String description;
  final bool isRandom;
  final DateTime startDate;
  final DateTime endDate;
  final int passScore;
  final bool isActive;

  const Exam({
    required this.examId,
    required this.name,
    required this.description,
    required this.isRandom,
    required this.startDate,
    required this.endDate,
    required this.passScore,
    required this.isActive,
  });

  factory Exam.fromJson(Map<String, dynamic> json) {
    return Exam(
      examId: json['examId'],
      name: json['name'],
      description: json['description'],
      isRandom: json['isRandom'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      passScore: json['passScore'],
      isActive: json['isActive'],
    );
  }
}
